hogben/utils.py

```diff
@@ -163,77 +157,182 @@ def fisher(qs: list[np.ndarray],
     "layer1" can be set to 2 using `layer1.thickness.importance = 2` or
     `layer1.thick.importance = 2` in `refnx` and `refl1d` respectively.
 
-    Args:
+    Attributes:
         qs: The Q points for each model.
```

Comment:
These aren't the attributes of the class, something between this and the __init__ docsting needs to be changed to show what needs to be input, and what the user facing methods/attributes are I think

Reply:
But they are the attributes as well, no? There's the attributes that are created upon `__init__`

```
        qs
        xi
        counts
        models
        step
```

And then there's the attributes that are defined as property:
```
fisher_information
min_eigenval
n
m
```
If anything, `n` and `m` are missing from the list. Although those may not be user facing strictly speaking.

All other attributes I'd consider user/facing ones, as they can directly be used by the user in real world application. After calculating the Fisher matrix, one can change one of these attributes (for example, try and see what one gets when changing the number of counts by calling `Fisher.counts = 200`, and then the updated information matrix can be retrieved directly again by `g = Fisher.information_matrix`.  So it makes sense to me to have all those attributes exposed to the user directly, as it allows to recalculate the information matrix without having to instantiate a whole new Fisher object.

At this point, there's no user-facing methods in this class. As both the minimum eigenvalue as well as the fisher information matrix are defined as property (which is a special kind of attribute), which is why I put them in the same list as the regular attributes.